src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs

```diff
@@ -35,6 +35,25 @@ public void It_fails_if_windows_target_platform_version_is_invalid()
                 .HaveStdOutContaining("NETSDK1137");
         }
 
+        [WindowsOnlyFact]
+        public void It_succeeds_if_windows_target_platform_version_has_trailing_zeros()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "ValidWindowsVersion",
+                IsSdkProject = true,
+                TargetFrameworks = "net5.0"
+            };
+            testProject.AdditionalProperties["TargetPlatformIdentifier"] = "windows"; 
+            testProject.AdditionalProperties["TargetPlatformVersion"] = "10.0.18362.0"; // We must set this manually because if we set it in the TFM we remove the trailing zeroes. 
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute()
+                .Should()
+                .Pass();
```

Comment:
I would also add to this test a check that the `TargetPlatformVersion` gets the trailing `.0` trimmed off.